torch/distributed/rpc/internal.py

```diff
@@ -113,15 +113,21 @@ def serialize(self, obj):
         # Ignore type error because dispatch_table is defined in third-party package
         p.dispatch_table[dist.rpc.RRef] = self._rref_reducer  # type: ignore[index]
 
-        # Install customized picklers.
-        for class_name in self._class_reducer_dict.keys():
-            p.dispatch_table[class_name] = self._class_reducer_dict[class_name]  # type: ignore[index]
-
-        # Add dispatch pickling for ScriptModule if needed.
+        # Add dispatch pickling for ScriptModule or its subclass.
         if isinstance(obj, torch.jit.ScriptModule):
```

Comment:
Doesn't simply removing this `if` statement and always registering the reducer in the dispatch table solve this problem? Do we actually need a separate entry for RecursiveScriptModule in the dispatch table? If we do need two separate entries, I guess we can just add them unconditionally and still remove this if-statement?